app/javascript/components/ActionItemForm.jsx

```diff
@@ -0,0 +1,403 @@
+    
+    }
+    handleOpen() {
+        this.setState({open: true});
+    }
+    handleClose() {
+        this.setState({ 
+        open: false,
+        internal: this.props.internal,
+        title: this.props.title
+      });
+    //handleInternalChange = 
+    if(this.state.type == 'edit') {
+      this.state.internal = this.props.internal,
+      this.state.title = this.props.title,
+      this.state.description = this.props.description
+    }
+  }
+  checkErrors = field => () => {
+    let errorMessage = '';
+    if (field === 'title') {
+      const { title } = this.state;
+      if (
+        title === '' ||
+        validator.isEmpty(title, { ignore_whitespace: true })
+      ) {
+        errorMessage = 'Title is required';
+      }
+    }
+    this.setState(prevState => ({
+      errors: { ...prevState.errors, [field]: errorMessage },
+    }));
+  };
+
+  handleChange = name => event => {
+    const { value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleInternalChange = name => event => {
+    this.setState(
+      { [name]: event.target.checked }
+    );
+  };
+
+  handleDescriptionChange = name => state => {
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit() {
+    const { type } = this.state;
+
+    let hasErrors = false;
+    Object.keys(this.state.errors).forEach(field => {
+      this.checkErrors(field)();
+      hasErrors = hasErrors || this.state.errors[field] !== '';
+    });
+
+    if (!hasErrors) {
+      if (type === 'create') {
+        const body = {
+          title: this.state.title,
+          description: this.state.description,
+          internal: this.state.internal,
+          participant_id: this.state.participant_id,
+        };
+        
+        apiPost('/api/case_notes', { case_note: body })
+          .then(() => window.location.reload())
+          .catch(error => console.error(error));
+      } else {
+        
+        const newTitle = this.state.title;
+        const newDescription = this.state.tempDescription;
+        const newInternal = this.state.internal;
+
+        this.setState({
+          title: newTitle,
+          description: newDescription,
+          internal: newInternal,
+        });
+  
+      const body = {
+        title: this.state.title,
+        description: this.state.description,
+        internal: this.state.internal,
+        participant_id: this.state.participant_id
+      };
+      apiPatch(`/api/case_note/${this.state.id}/`, {case_note: body}).then(() => window.location.reload())
+      .catch(error => console.error(error));
+    }
+  }
+}
+button = () => {
+  let ret;
+  if (this.state.display == 'plus') {
+    ret = (
+      <button onClick={this.handleOpen} className="plus-button">
+        +
+      </button>
+    );
+  } else if (this.state.type === 'create') {
+    ret = (
+      <Button
+        className="primary-button"
```

Comment:
It seems kind of weird to have className="primary-button" and have the color of the button be "secondary". I think this might be a problem with casenotes/paperworks that we might have to fix too. Maybe something to do for midpoint hackathon